frontend/test/metabase/scenarios/binning/qb-implicit-joins.cy.spec.js

```diff
@@ -0,0 +1,86 @@
+import { restore, popover } from "__support__/e2e/cypress";
+
+describe("scenarios > binning > from a saved QB question using implicit joins", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+
+    cy.intercept("POST", "/api/dataset").as("dataset");
+  });
+
+  context("via simple question", () => {
+    beforeEach(() => {
+      cy.visit("/question/1");
+      cy.findByText("Summarize").click();
+    });
+
+    it("should work for time series", () => {
+      cy.findByTestId("sidebar-right").within(() => {
+        openPopoverFromDefaultBucketSize("Birth Date", "by month");
+      });
+
+      popover().within(() => {
+        cy.findByText("Year").click();
+      });
+
+      waitAndAssertOnRequest("@dataset");
+
+      cy.findByText("Count by User → Birth Date: Year");
+
+      // The default chosen "visualization" is table
+      cy.findByText("1958");
+      cy.findByText("313");
+    });
+
+    it("should work for number", () => {
+      cy.findByTestId("sidebar-right").within(() => {
+        openPopoverFromDefaultBucketSize("Price", "Auto bin");
+      });
+
+      popover().within(() => {
+        cy.findByText("10 bins").click();
+      });
+
+      waitAndAssertOnRequest("@dataset");
+
+      cy.findByText("Count by Product → Price: 10 bins");
+      cy.findByText("196");
+    });
+
+    it.skip("should work for longitude", () => {
+      cy.findByTestId("sidebar-right").within(() => {
+        openPopoverFromDefaultBucketSize("Longitude", "Auto bin");
+      });
+
+      popover().within(() => {
+        // Test fails at this point (UI inconsistency).
+        // It simply gives `10°` as the option, which is different than in any other longitude binning scenario.
```

Comment:
Could this be changed, so it works with the inconsistency - we have #16674 open for addressing the inconsistencies.
It's far more important that we don't break the functionality than a tiny string inconsistency.

Reply:
I was thinking about it, but then what happens when we fix 16674? We'd need to update this test as well.

My logic was - if there's a workaround the issue, use it and "silently ignore" the fact that something is incorrect it the goal is to test far more important thing like rendering.
OTOH, if we have to use incorrect value just to make the test work, it creates a new technical debt and an overhead because it'll need to be updated in the future.

I can go either way - just trying to explain my logic.

Reply:
What's worse: That binning stops working, or that some string shows `10°` instead of `Binning by 10°` (or whatever it will be called) ?
If you already have a test somewhere that checks that this type of binning works, then great - ignore this - but otherwise it's not skip a test because of some text inconsistency, which is a P3 UX.

Reply:
Got it.
This particular edge case **was not** covered by the correctness test #16630. I was testing values on the regular table, because I couldn't suspect that those change based on whether the table is implicitly/explicitly joined. Now that we have these edge cases documented, I'll update the correctness tests accordingly and will update this one to check the more important thing.

Reply:
This is not Correctness - if you clicked `10°` but the result was e.g. `50°`, then it's a Correctness issue.
This is just a string prepended with some text. Just change the test to only look for the degrees inside of popover, then it would work - no matter what the text might be changed to.
I wouldn't call it an edge-case either. The use of implicit (or explicit) fields are very central to Metabase.